alf/environments/suite_highway.py

```diff
@@ -0,0 +1,144 @@
+# limitations under the License.
+"""Suite for loading highway environments.
+    Installation:
+    pip install git+https://github.com/eleurent/highway-env
+"""
+
+import collections
+import gin
+import gym
+import gym.spaces
+
+from alf.environments import suite_gym, alf_wrappers, gym_wrappers, process_environment
+
+try:
+    import highway_env
+except ImportError:
+    highway_env = None
+
+
+def is_available():
+    return highway_env is not None
+
+
+class VectorizeObservation(gym_wrappers.BaseObservationWrapper):
+    def transform_space(self, observation_space):
+        return gym.spaces.Box(
+            low=-observation_space.low.ravel(),
+            high=observation_space.high.ravel())
+
+    def transform_observation(self, observation):
+        return observation.ravel()
+
+
+class RemoveActionEnvInfo(gym.Wrapper):
+    """Remove action from EnvInfo if exist
+    """
+
+    def step(self, action):
+        obs, reward, done, env_info = self.env.step(action)
+        env_info.pop('action', None)
+        return obs, reward, done, env_info
+
+
+@gin.configurable
+def load(environment_name,
+         env_id=None,
+         discount=1.0,
+         max_episode_steps=None,
+         gym_env_wrappers=(),
+         alf_env_wrappers=(),
+         env_config=None):
+    """Loads the selected environment and wraps it with the specified wrappers.
+
+    Note that by default a TimeLimit wrapper is used to limit episode lengths
+    to the default benchmarks defined by the registered environments.
+
+    Args:
+        environment_name (str): Name for the environment to load.
+        env_id (int): (optional) ID of the environment.
+        discount (float): Discount to use for the environment.
+        max_episode_steps (int): If None the max_episode_steps will be set to the
+            default step limit defined in the environment's spec. No limit is applied
+            if set to 0 or if there is no max_episode_steps set in the environment's
+            spec.
+        gym_env_wrappers (Iterable): Iterable with references to gym_wrappers
+            classes to use directly on the gym environment.
+        alf_env_wrappers (Iterable): Iterable with references to alf_wrappers
+            classes to use on the ALF environment.
+        env_config (dict|None): a dictionary for configuring some aspects of the
+            environment. If is None, the default configuration will be used.
+            Please refer to the ``default_env_config`` below for
+            an example config and the doc for more details:
+            https://highway-env.readthedocs.io/en/latest/user_guide.html
+
+    Returns:
+        An AlfEnvironment instance.
+    """
+    assert environment_name in {
+        "highway-v0", "merge-v0", "roundabout-v0", "intersection-v0",
+        "parking-v0"
+    }, "wrong highway environment name"
+
+    gym_spec = gym.spec(environment_name)
+    gym_env = gym.make(environment_name)
```

Comment:
Does the env itself use gym.make to wrap with max_steps?

Reply:
The original env does not differentiate terminations due to max_steps or others.
Now correctly handle max_steps in a way similar to suite_safety_gym.

Reply:
I mean by calling "gym.make", it probably already wrap an incorrect time_limit wrapper. Should use "gym_spec.make" instead.

Reply:
Right, I changed to use ``gym.spec.make`` instead of ``gym.make``in the second commit because of this.